Tests/test_file_gimpgradient.py

```diff
@@ -1,122 +1,128 @@
-        self.assertEqual(ret, 0.25)
 
-    def test_linear_pos_le_small_middle(self):
-        # Arrange
-        middle = 1e-11
-        pos = 1e-12
+def test_linear_pos_le_small_middle():
+    # Arrange
+    middle = 1e-11
+    pos = 1e-12
 
-        # Act
-        ret = GimpGradientFile.linear(middle, pos)
+    # Act
+    ret = GimpGradientFile.linear(middle, pos)
 
-        # Assert
-        self.assertEqual(ret, 0.0)
+    # Assert
+    assert ret == 0.0
 
-    def test_linear_pos_gt_middle(self):
-        # Arrange
-        middle = 0.5
-        pos = 0.75
 
-        # Act
-        ret = GimpGradientFile.linear(middle, pos)
+def test_linear_pos_gt_middle():
+    # Arrange
+    middle = 0.5
+    pos = 0.75
 
-        # Assert
-        self.assertEqual(ret, 0.75)
+    # Act
+    ret = GimpGradientFile.linear(middle, pos)
 
-    def test_linear_pos_gt_small_middle(self):
-        # Arrange
-        middle = 1 - 1e-11
-        pos = 1 - 1e-12
+    # Assert
+    assert ret == 0.75
 
-        # Act
-        ret = GimpGradientFile.linear(middle, pos)
 
-        # Assert
-        self.assertEqual(ret, 1.0)
+def test_linear_pos_gt_small_middle():
+    # Arrange
+    middle = 1 - 1e-11
+    pos = 1 - 1e-12
 
-    def test_curved(self):
-        # Arrange
-        middle = 0.5
-        pos = 0.75
+    # Act
+    ret = GimpGradientFile.linear(middle, pos)
 
-        # Act
-        ret = GimpGradientFile.curved(middle, pos)
+    # Assert
+    assert ret == 1.0
 
-        # Assert
-        self.assertEqual(ret, 0.75)
 
-    def test_sine(self):
-        # Arrange
-        middle = 0.5
-        pos = 0.75
+def test_curved():
+    # Arrange
+    middle = 0.5
+    pos = 0.75
 
-        # Act
-        ret = GimpGradientFile.sine(middle, pos)
+    # Act
+    ret = GimpGradientFile.curved(middle, pos)
 
-        # Assert
-        self.assertEqual(ret, 0.8535533905932737)
+    # Assert
+    assert ret == 0.75
 
-    def test_sphere_increasing(self):
-        # Arrange
-        middle = 0.5
-        pos = 0.75
 
-        # Act
-        ret = GimpGradientFile.sphere_increasing(middle, pos)
+def test_sine():
+    # Arrange
+    middle = 0.5
+    pos = 0.75
 
-        # Assert
-        self.assertAlmostEqual(ret, 0.9682458365518543)
+    # Act
+    ret = GimpGradientFile.sine(middle, pos)
 
-    def test_sphere_decreasing(self):
-        # Arrange
-        middle = 0.5
-        pos = 0.75
+    # Assert
+    assert ret == 0.8535533905932737
 
-        # Act
-        ret = GimpGradientFile.sphere_decreasing(middle, pos)
 
-        # Assert
-        self.assertEqual(ret, 0.3385621722338523)
+def test_sphere_increasing():
+    # Arrange
+    middle = 0.5
+    pos = 0.75
 
-    def test_load_via_imagepalette(self):
-        # Arrange
-        from PIL import ImagePalette
+    # Act
+    ret = GimpGradientFile.sphere_increasing(middle, pos)
 
-        test_file = "Tests/images/gimp_gradient.ggr"
+    # Assert
+    assert round(abs(ret - 0.9682458365518543), 7) == 0
```

Comment:
The table just above https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual doesn't list `abs` as part of the `assertAlmostEqual` operation?

Reply:
For clarity:

```diff
-        self.assertAlmostEqual(ret, 0.9682458365518543)
+    assert round(abs(ret - 0.9682458365518543), 7) == 0

```

---

You're right, that table says `assertAlmostEqual` checks

```python
round(a-b, 7) == 0
```

but the stdlib does

```python
round(abs(second-first), places) == 0
```

https://github.com/python/cpython/blob/276eb67c29d05a93fbc22eea5470282e73700d20/Lib/unittest/case.py#L870

essentially

```python
round(abs(b-a), 7) == 0
```

It can be mathematically proved `abs(b-a) == abs(a-b)`, but I'm not really sure if the `abs` is strictly necessary. At least in this case:

```python
>>> round(0.000000009, 7)
0.0
>>> round(-0.000000009, 7)
-0.0
>>> round(0.000000009, 7) == 0
True
>>> round(-0.000000009, 7) == 0
True
```